### DISCUSSION TITLE
Access permission error with Xcode

### DISCUSSION THREAD

**Discussion by asnagni** [ccache/ccache#1425] [Support]
Hi guys,
    To improve our build time with CMAKe, I was looking for a solution and I found this article (https://crascit.com/2016/04/09/using-ccache-with-cmake/) and I'm trying to use this solution for my Xcode project and I'm having a permission issue. This is the error that I'm having:

```
ccache: error: execute_noreturn of /Applications/Xcode.app/Contents/Developer/usr/bin/ failed: Permission denied
Command CompileC failed with a nonzero exit code
```

Hi could not find the CompileC program to change the permission with the chmod command.
Would you have any idea on how to fix this issue?
Thank you
Stay safe

### How to reproduce ###

1.  install the the two scripts
2. Add the modification at the top of your cmake file
3. build

### Expected behavior ###

I would expect to see the compilation completed and be faster than without using cache

### Environment ###

ccache version 4.9.1

Xcode 15.3
macOS Sonoma 14.4.1

**Comment by res0nance** [comment]
This is an issue with your environment and cmake, nothing to do with ccache itself.

**Reply by kevin--** [reply]
did you determine what is the issue with the environment? I'm experiencing the same issue.
